ext/sockets/sockets.c

```diff
@@ -2456,6 +2512,246 @@ PHP_FUNCTION(socket_export_stream)
 }
 /* }}} */
 
+/* {{{ proto resource addrinfo socket_addrinfo_lookup(string hostname[, mixed service, array hints])
+   Gets array with contents of getaddrinfo about the given hostname. */
+PHP_FUNCTION(socket_addrinfo_lookup)
+{
+	zend_string *hostname, *key;
+	zval *hint, *service, *zhints = NULL;
+
+	struct addrinfo hints, *result, *rp, *res;
+
+	memset(&hints, 0, sizeof(hints));
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "S|za", &hostname, &service, &zhints) == FAILURE) {
+		RETURN_NULL();
+	}
+
+	if (zhints) {
+		ZEND_HASH_FOREACH_STR_KEY_VAL(Z_ARRVAL_P(zhints), key, hint) {
+			if (key) {
+				if (strcmp(ZSTR_VAL(key), "ai_flags") == 0) {
+					hints.ai_flags = Z_LVAL_P(hint);
+				} else if (strcmp(ZSTR_VAL(key), "ai_socktype") == 0) {
+					hints.ai_socktype = Z_LVAL_P(hint);
+				} else if (strcmp(ZSTR_VAL(key), "ai_protocol") == 0) {
+					hints.ai_protocol = Z_LVAL_P(hint);
+				} else if (strcmp(ZSTR_VAL(key), "ai_family") == 0) {
+					hints.ai_family = Z_LVAL_P(hint);
+				}
+			}
+		} ZEND_HASH_FOREACH_END();
+	}
+
+	convert_to_string(service);
```

Comment:
getaddrinfo()'s second arg is a char*, not an int.  So if someone was to give us a long value, we need to convert it to a char* for use in getaddrinfo()